client/packages/system/src/Patient/api/hooks/document/usePatients.ts

```diff
@@ -1,7 +1,20 @@
-import { useQuery } from '@openmsupply-client/common';
+import { useMutation, useQuery } from '@openmsupply-client/common';
 import { ListParams } from '../../api';
 import { usePatientApi } from '../utils/usePatientApi';
 
+export const usePatientsPromise = () => {
+  const api = usePatientApi();
+
+  return useMutation(async (query: ListParams) => {
+    const patients = await api.get.list(query);
+
+    return {
+      nodes: patients.nodes,
+      totalCount: patients.totalCount,
+    };
+  });
```

Comment:
You can just return the `patients` object here(?) Think this is an error/unnecessary in the other hook as well...

Reply:
I've refactored to both return the full object. However the refactor for the program hook was a bit more complicated and moved the filtering for latest status event into the IdGenerator component. Was this what you had envisioned?

Reply:
Thanks, I meant you can just do `return patients`, i.e. just reuse the same object you got from the api.

You can remove the whole, `const latestNodes = (nodes: ProgramEnrolmentRowFragment[]) => {...}`. This is a left over from some previous use-case and shouldn't be in there anymore... Since the hook is only used in IdGenerator it should be save to remove it. I will test it once you are done.